chainer/links/normalization/layer_normalization.py

```diff
@@ -0,0 +1,77 @@
+from chainer import initializers
+from chainer import link
+
+from chainer.functions.array import broadcast
+from chainer.functions.math import sqrt
+from chainer.functions.math import square
+from chainer.functions.math import sum
+from chainer.links.connection import scale
+
+
+class LayerNormalization(link.Chain):
+
+    """Layer normalization layer on outputs of linear functions.
+
+    This link implements a "layer normalization" layer
+    which normalizes the input units by statistics
+    that are computed along the second axis,
+    scales and shifts them with :class:`~chainer.links.Scale`.
+
+
+    Args:
+        size (int): Size of input units.
+        eps (float): Epsilon value for numerical stability of normalization.
+        initial_gumma (~chainer.Initializer): Initializer for scaling vector.
+            If ``None``, then the vector is initialized
+            by :class:`~chainer.initializers.HeNormal`.
+            If a scalar, the vectors are filled by it.
+            If ``numpy.ndarray``, the vectors are set by it.
+        initial_beta (~chainer.Initializer): Initializer for shifting vector.
+            If ``None``, then the vector is initialized
+            by :class:`~chainer.initializers.HeNormal`.
+            If a scalar, the vectors are filled by it.
+            If ``numpy.ndarray``, the vectors are set by it.
+
+    Attributes:
+        scale (~chainer.links.Scale): Layer to scale and shift
+            units after normalization.
+        eps (float): Epsilon value for numerical stability.
+
+    See: `Layer Normalization <https://arxiv.org/abs/1607.06450>`_
+    """
+
+    def __init__(self, size, eps=1e-6, initial_gamma=None, initial_beta=None):
+        super(LayerNormalization, self).__init__(
+            scale=scale.Scale(axis=1, W_shape=(size, ), bias_term=True),
+        )
+        if initial_gamma is None:
```

Comment:
I suggest to use `chainer.utils.experimental.py` to mark this class as having an experimental API.  This is a new Chainer feature that was just merged into the master branch yesterday (12/5).  This will allow us to the flexibility to continue modifying the API of new Chainer features without having to worry so much about getting the API perfect the fist time. You can use the "experimental feature API" by calling `experimental('chainer.links.normalization.layer_normalization.py')` in the `__init__()` method which will mark the entire LayerNormalization class as having an experimental API.